crates/ty_ide/src/completion.rs

```diff
@@ -40,6 +42,27 @@ pub fn completion(db: &dyn Db, file: File, offset: TextSize) -> Vec<Completion<'
     completions.sort_by(compare_suggestions);
     completions.dedup_by(|c1, c2| c1.name == c2.name);
     completions
+        .into_iter()
+        .map(|completion| {
+            let definition = DefinitionsOrTargets::from_ty(db, completion.ty);
+            let documentation = definition.and_then(|def| def.docstring(db));
+            DetailedCompletion {
+                inner: completion,
+                documentation,
+            }
+        })
+        .collect()
+}
+
+pub struct DetailedCompletion<'db> {
+    pub inner: Completion<'db>,
+    pub documentation: Option<Docstring>,
```

Comment:
Should we make this a field on `Completion` instead that defaults to `None`: Given that `completion` is the only method constructing the completions